### PR TITLE
Update common app control for common privacy setting guide

### PR DESCRIPTION
**Pull request by yl33**
### Change Description ###

Add description of new common appcontrol operation APP_CONTROL_OPERATION_PRIVACY_SETTING_GUIDE

### API Changes ###

 - ACR: ACR-1227

### CODE REVIEW

File: docs/application/native/guides/app-management/common-appcontrol.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 An application control provides functions for launching other applications with a specific operation, URI, MIME type, and extra data. The requesting application can get a result back from the launched application. This topic introduces the common application controls that you can use.
 
-> **Note**  
+> **Note**
```
[the remark this answers is not in the document]
Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: done

File: docs/application/native/guides/app-management/common-appcontrol.md [hunk 2]
```diff
@@ -211,7 +211,7 @@ For example: `file://<media storage path>/file.vcs`
 
 #### MIME Type
 
-- `application/vnd.tizen.calendar`  
+- `application/vnd.tizen.calendar`
```
[the remark this answers is not in the document]
Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: done

File: docs/application/native/guides/app-management/common-appcontrol.md [hunk 3]
```diff
@@ -637,7 +637,7 @@ To display a specified contact from a vcard file, use the `file:` URI. To displa
 
 #### MIME Type
 
-- `application/vnd.tizen.contact`  
+- `application/vnd.tizen.contact`
```
[the remark this answers is not in the document]
Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: done

**Comment by mijin85cho** [comment]
Every comments from me are fixed. Thanks for the update. : )

Also, please check the comments which will be created by SRI-B,
when they finish reviewing this PR.

**Comment by mijin85cho** [comment]
Hello, @annie-samsung
This comment to remind you of this PR.
Would you please assign a person to this PR?

File: docs/application/native/guides/app-management/common-appcontrol.md [hunk 4]
```diff
@@ -2239,6 +2239,83 @@ vpn_appcontrol_result_cb(app_control_h request, app_control_h reply, app_control
...
     dlog_print(DLOG_INFO, LOG_TAG, "Result: %s", result_txt);
 }
+```
+## Privacy setting guide popup
+
+The Privacy setting guide popup application control is supported only in wearable applications since Tizen 5.0.
+
+### Shows privacy related common guide popup
+
+To show common privacy setting guide popup, use the `APP_CONTROL_OPERATION_PRIVACY_SETTING_GUIDE` operation. This operation helps you to guide users to change privacy privilege settings. You need to check permissions to see if the user actually granted requested permissions after using this application control.
```
Comment (Reni-90) [inline docs/application/native/guides/app-management/common-appcontrol.md]: Suggestion:

Rephrase the last sentence to:
After using this application control, check permissions to verify whether the user is granted with the requested permissions.

Please verify that the technical meaning is not changed.

File: docs/application/native/guides/app-management/common-appcontrol.md [hunk 5]
```diff
@@ -1163,8 +1163,8 @@ The input types are grouped into the following MIME types. Therefore, if you spe
...
+| `APP_CONTROL_DATA_INPUT_REPLY_TYPE` | The reply type. This key must be passed as a string.<br>This key allows caller application to know the user has entered though which input type of Input Delegator.<br><br> You can decide how to use extra output  according to the result  of this key. If the  result is `input_audio`   or `input_image`, you  must get the path of the result value through     `APP_CONTROL_DATA_PATH`.<br><br>The available values    are:     <br>  -   `input_voice`:     Receive the result     as voice <br>     -   `input_emoticon`:  Receive the result   as an emoticon  <br>-   `input_keyboard`:      Receive the result    as keyboard input  <br> -   `input_reply`:          Receive the result    as reply input     <br> -   `input_image`:     Receive the result    as an image  <br> -   `input_audio`:        Receive the result    as audio  | This key is optional and is supported only in wearable applications since Tizen 4.0. |
```
[the remark this answers is not in the document]
Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Input delegator is not my responsibility so I can't decide that.

File: docs/application/native/guides/app-management/common-appcontrol.md [hunk 6]
```diff
@@ -2239,6 +2239,83 @@ vpn_appcontrol_result_cb(app_control_h request, app_control_h reply, app_control
     app_control_get_extra_data(reply, APP_CONTROL_DATA_TEXT, &result_txt);
     dlog_print(DLOG_INFO, LOG_TAG, "Result: %s", result_txt);
 }
+```
+## Privacy setting guide popup
+
+The Privacy setting guide popup application control is supported only in wearable applications since Tizen 5.0.
```
[the remark this answers is not in the document]
Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Done

File: docs/application/native/guides/app-management/common-appcontrol.md [hunk 7]
```diff
@@ -2239,6 +2239,83 @@ vpn_appcontrol_result_cb(app_control_h request, app_control_h reply, app_control
...
+
+### Shows privacy related common guide popup
+
+To show common privacy setting guide popup, use the `APP_CONTROL_OPERATION_PRIVACY_SETTING_GUIDE` operation. This operation helps you to guide users to change privacy privilege settings. You need to check permissions to see if the user actually granted requested permissions after using this application control.
+
+**Figure: Privacy setting guide popup**
+
+![Guide popup for app launching](./media/common_appcontrol_privacy_setting_guide_open.png)![Guide popup for using privacy privileged function](./media/common_appcontrol_privacy_setting_guide_use.png)![Required privacy list shown in popup](./media/common_appcontrol_privacy_setting_guide_privacy.png)
+
+#### Operation
```
[the remark this answers is not in the document]
Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Done

File: docs/application/native/guides/app-management/common-appcontrol.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Reni-90) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Ok. Keep it as delegator only and rephrase the sentence if the technical meaning is not changed.

File: docs/application/native/guides/app-management/common-appcontrol.md [hunk 8]
```diff
@@ -2239,6 +2239,83 @@ vpn_appcontrol_result_cb(app_control_h request, app_control_h reply, app_control
...
+- `http://tizen.org/appcontrol/operation/setting/guide_privacy_setting` (in `.c` files and manifest file)
+- `APP_CONTROL_OPERATION_PRIVACY_SETTING_GUIDE` (in `.c` files only)
+
+#### Extra Input
+
+| Key                     | Description                              | Note                   |
+|-------------------------|------------------------------------------|------------------------|
+| `APP_CONTROL_DATA_TYPE` | The guide message type. The available values are `open` (for application launching) and `use` (for using function).  This key must be passed as a string. | This key is mandatory. |
+| `APP_CONTROL_DATA_FEATURE` | The feature or function name to show. This key must be passed as a string. |   |
+| `APP_CONTROL_DATA_PRIVILEGES` | The required privacy privileges to show. This key must be passed as an array of string.| The required privacy privileges to show. This key must be passed as an array of string. |
```
[the remark this answers is not in the document]
Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Modified note of 'APP_CONTROL_DATA_PRIVILEGES'. 

For description, I added "to show" because the app developer can select some of the privacy privileges. For example, A, B, and C are required privacy privileges of the application but the developer can decide to request user only A and B for now. I just added it to say "don't need to add all privacy privileges of the application". Should it be modified? Need your opinion.

File: docs/application/native/guides/app-management/common-appcontrol.md [hunk 9]
```diff
@@ -2239,6 +2239,83 @@ vpn_appcontrol_result_cb(app_control_h request, app_control_h reply, app_control
...
+
+- `http://tizen.org/appcontrol/operation/setting/guide_privacy_setting` (in `.c` files and manifest file)
+- `APP_CONTROL_OPERATION_PRIVACY_SETTING_GUIDE` (in `.c` files only)
+
+#### Extra Input
+
+| Key                     | Description                              | Note                   |
+|-------------------------|------------------------------------------|------------------------|
+| `APP_CONTROL_DATA_TYPE` | The guide message type. The available values are `open` (for application launching) and `use` (for using function).  This key must be passed as a string. | This key is mandatory. |
+| `APP_CONTROL_DATA_FEATURE` | The feature or function name to show. This key must be passed as a string. |   |
```
[the remark this answers is not in the document]
Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Removed "to show".

File: docs/application/native/guides/app-management/common-appcontrol.md (hunk 4 above)
Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Not the user who's getting a permission. It's the application. User can grant a permission to the application to perform specific function. How about "After using this application control, check permissions to verify whether the user granted the requested permissions." ? Need your opinion.

Comment (Reni-90) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Perfect!!
Sorry, I misinterpreted. Please go ahead and rephrase the sentence to "After using this application control, check permissions to verify whether the user granted the requested permissions."

Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Done.

File: docs/application/native/guides/app-management/common-appcontrol.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (Reni-90) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Ok. 
Then can we rephrase it to: 
The required privacy privileges that are selected.
OR
The required privacy privileges, which are displayed.
OR
The required privacy privileges that are requested.
OR
The required and requested privacy privileges.
OR
You can rephrase it better.But, we will avoid "to show". Need your opinion.

[the remark this answers is not in the document]
Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: You mean.. you're avoiding ending a sentence with "to OOO" form? If it's not, "to display" or "to request" would be OK.
Or I want to use 'The required privacy privileges that are requested.'.
Need your opinion.

[the remark this answers is not in the document]
Comment (Reni-90) [reply docs/application/native/guides/app-management/common-appcontrol.md]: As a reader, when I read 'The required privacy privileges to show' or 'The required privacy privileges to display' confuses me and breaks the flow for me. Because I feel like something is to show or something is to get displayed.
If you use:
'The required privacy privileges that are requested.' Then your description " For example, A, B, and C are required privacy privileges of the application but the developer can decide to request user only A and B for now." goes well with this.

Hence, I feel we should stick to something like 'The required privacy privileges that are requested.'.
Need your opinion.

[the remark this answers is not in the document]
Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Modified as 'The required privacy privileges that are requested.'. Thank you for kind explanation.

[the remark this answers is not in the document]
Comment (Reni-90) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Thank you for understanding and updating the changes.

[the remark this answers is not in the document]
Comment (mijin85cho) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Hello. Reni-90, and yl33

I can see you had discussion about this "The required privacy privileges that are requested." 

However unfortunately at the first glance, I couldn't get the meaning of the sentence... As it sounds like "Asked privacy privileges asked.". The discussion history here barely made me understand, but the users out there can't read this discussion.

So Reni-90, I hope we get better sentence than 'The required privacy privileges that are requested.`. 
Would you please think about it more to clarify the meaning of "requested privacy privilege to an user, among required privileges"?
Or I think we can focus on the popup description itself so to change the sentence like "The  privacy privileges to be requested on popup", not comparing whole required privileges and requested privileges.

Regards

**Comment by mijin85cho** [comment]
Hello, @Reni-90 

This is a reminder comment .
Would you please check my previous comment and give your opinion about it?

---
Hello. Reni-90, and yl33

I can see you had discussion about this "The required privacy privileges that are requested."

However unfortunately at the first glance, I couldn't get the meaning of the sentence... As it sounds like "Asked privacy privileges asked.". The discussion history here barely made me understand, but the users out there can't read this discussion.

So Reni-90, I hope we get better sentence than 'The required privacy privileges that are requested.`.
Would you please think about it more to clarify the meaning of "requested privacy privilege to an user, among required privileges"?
Or I think we can focus on the popup description itself so to change the sentence like "The privacy privileges to be requested by popup", not comparing whole required privileges and requested privileges.

Regards

**Comment by Reni-90** [comment]
Sorry for the delay.
Can we rephrase as:
The requested privileges among the required privacy privileges.
OR
The requested privileges among the required privacy privileges that gets displayed.
Need your opinion.

**Comment by yl33** [comment]
Can I add a note for it? Want to place after following sentence: 'You need to check permissions to see if the user actually granted requested permissions after using this application control.'.

This app control's launch_mode is 'group'. When calling this app control, the caller should care about the app control's life cycle.

**Comment by mijin85cho** [comment]
Hello, yl33

Yes, of course you can add a note. Please update the PR and ask Reni-90 to check the sentence.
Also, please check Reni-90's suggestion above.

Regards

**Comment by yl33** [comment]
Hello. mijin85cho, and Reni-90.
How about "The required privacy privileges that gets displayed"? I want to shorten Reni-90's suggestion: "The requested privileges among the required privacy privileges that gets displayed."

**Comment by mijin85cho** [comment]
Hello, @yl33 
I 'm ok with your suggestion. How about you, @Reni-90 ?

**Comment by yl33** [comment]
Hello, @Reni-90 and @mijin85cho 
I update the description and also added the note I mentioned above. Please check whether the note is OK.

File: docs/application/native/guides/app-management/common-appcontrol.md [hunk 10]
```diff
@@ -2246,7 +2246,7 @@ From Tizen 5.0, the privacy setting guide popup application control is supported
 
 ### Shows privacy related common guide popup
 
-To show common privacy setting guide popup, use the `APP_CONTROL_OPERATION_PRIVACY_SETTING_GUIDE` operation. This operation helps you to guide users to change privacy privilege settings. After using this application control, check permissions to verify whether the user granted the requested permissions.
+To show common privacy setting guide popup, use the `APP_CONTROL_OPERATION_PRIVACY_SETTING_GUIDE` operation. This operation helps you to guide users to change privacy privilege settings. After using this application control, check permissions to verify whether the user granted the requested permissions. This app control's launch_mode is 'group'. When calling this app control, the caller should care about the app control's life cycle.
```
Comment (Reni-90) [inline docs/application/native/guides/app-management/common-appcontrol.md]: For, "After using this application control, check permissions to verify whether the user granted the requested permissions."
Please change it to:
"After using this application control, check permissions to verify whether the user has granted the requested permissions."


This line: "This app control's launch_mode is 'group'. When calling this app control, the caller should care about the app control's life cycle."
can we rephrase it to:
The launch mode of this application control is *group*. The launch mode of this application control is *group*. While calling this control, you must be aware about the app control life cycle.

I have a question here:
About what the user must be aware?
What is the issue caused if the caller is not paying attention to the life cycle?

Comment (mijin85cho) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Hello, Reni-90

I think we should keep `launch_mode` in this form as yl33 wrote.
Please refer [here](https://github.com/Samsung/tizen-docs/blob/master/docs/application/native/guides/app-management/app-controls.md#application-group-management). In the article, it is also guiding [how to set launch mode in the manifest file](https://github.com/Samsung/tizen-docs/blob/master/docs/application/tizen-studio/native-tools/manifest-text-editor.md) like below.

image

Also, I hope [this guide](https://github.com/Samsung/tizen-docs/blob/master/docs/application/native/tutorials/feature/best-practice-battery.md#app_battery_lifecycle) or [this one](https://github.com/Samsung/tizen-docs/blob/master/docs/application/native/guides/app-management/ui-app/efl-ui-app.md#application-states-and-transitions) can help you to understand.

However, is it common to use "aware about" in the guide?
Also, the same sentence is duplicated in your review result.
"*The launch mode of this application control is group.* The launch mode of this application control is group. While calling this control, you must be aware about the app control life cycle."

Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: This app control is launched as group with caller app. It means that the app control will not be terminated until the explicit sign of termination is given. Explicit signs are clicking h/w back key, clicking buttons on the app control, terminated by caller, or so. 

For example, user can press home key while the app control is launched and the app is lowered. Of course the app control goes down with the app. In recent app menu, the caller app's topmost screen is the app control. And when the app is resumed, app might try to call app control again thinking the app control is terminated by the home key input because they didn't launch the app control as group mode explicitly. If that kind of call occurs again and again then the app control process can be multiply and never terminate until the caller die. SO, I mentioned it on application control description to notify app developers can manage it properly.

Comment (Reni-90) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Yes we should keep launch_mode in this form as yl33 wrote. I am sorry.
I was rephrasing it in a word file, so got it wrong.
I was trying to write :
The launch_mode of this application control is group. While calling this control, you must be aware about the app control life cycle."

Comment (mijin85cho) [reply docs/application/native/guides/app-management/common-appcontrol.md]: @Reni-90 
In fact, I'm familiar with "aware of", but not so with "aware about".
So "aware about" sounds tough to me, thus I can't sure with it..

Also I include the search result about the "aware about" and "aware of"
https://forum.wordreference.com/threads/aware-of-aware-about.911804/
https://www.thesaurus.com/browse/be%20aware%20of
https://www.italki.com/question/368407

Would you please confirm that it is really ok, to use "aware about", not "aware of"?

Comment (Reni-90) [reply docs/application/native/guides/app-management/common-appcontrol.md]: @mijin85cho 
Thank you so much for pointing out. A new learning for me as well. 
I think 'aware of' is better than 'aware about'.
So, will rephrase to:
The launch_mode of this application control is group. While calling this control, you must be aware of the app control life cycle."

Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: Updated reviewed phrases.

Comment (Reni-90) [reply docs/application/native/guides/app-management/common-appcontrol.md]: @yl33 
Thank you.

File: docs/application/native/guides/app-management/common-appcontrol.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Reni-90) [reply docs/application/native/guides/app-management/common-appcontrol.md]: @yl33 
Can we rephrase the sentence:
This key must be passed as a string.<br>This key allows caller application to know the user has entered though which input type of Input Delegator.

to

The reply type allows caller application to identify through which input type of input delegator the user entered. This key must be passed as a string.

Please verify the technical meaning is not changed.

[the remark this answers is not in the document]
Comment (mijin85cho) [reply docs/application/native/guides/app-management/common-appcontrol.md]: @Reni-90 Hello, Reni. I have a question for this update.

I have thought that to write a guide document. using easy words & sentence structure is important, for the clear understanding of users. So I'm not sure about the reason why this sentence is modified. Would you please let me know?

[the remark this answers is not in the document]
Comment (Reni-90) [reply docs/application/native/guides/app-management/common-appcontrol.md]: @mijin85cho 
 When i read this sentence, as a user: 
This key allows caller application to know the user has entered though which input type of Input Delegator.
I have to read it twice to understand the meaning, so I changed it to:
The reply type allows caller application to identify through which input type of input delegator the user entered. 

We can also modify the sentence as:
The reply type.  This key must be passed as a string.
This key allows caller application to identify through which input type of input delegator the user entered. 

My intention was to make it simple. Does it changes the technical meaning?

**Comment by yl33** [comment]
@mijin85cho My mistake. Placed it between 'multimedia' and 'system settings'. Thank you for the notice.

**Comment by mijin85cho** [comment]
@yl33  Thank you : D

File: docs/application/native/guides/app-management/common-appcontrol.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (mijin85cho) [reply docs/application/native/guides/app-management/common-appcontrol.md]: @Reni-90 
To me, at first glance the modification looks less effective as it is changing the structure back and forth, and changing "know" to "identify". However if the suggested sentence is easier to understand at once, I don't have objection.

Simple question : In two suggested sentences, different tenses are used. "user entered" and "user has entered"
I'm not sure why the tense is changed between the similar sentences. Would you please explain me?

 - *The original* "This key allows caller application to know the user has entered though which input type of Input Delegator."

 - *Suggested 1* "The reply type allows caller application to identify through which input type of input delegator the **user entered**."

 - *Suggested 2* "This key allows caller application to identify through which input type of input delegator the **user has entered**."

[the remark this answers is not in the document]
Comment (yl33) [reply docs/application/native/guides/app-management/common-appcontrol.md]: @mijin85cho  and @Reni-90 ,
Input delegator is not my module. I don't know rephrasing the sentence in this PR is the right thing.

[the remark this answers is not in the document]
Comment (mijin85cho) [reply docs/application/native/guides/app-management/common-appcontrol.md]: @Reni-90 As @yl33 said, please don't ask update for the Input Delegator as it is not yl33's responsibility. Keep current status.

**Comment by mijin85cho** [comment]
@Reni-90 Please check if everything is ok, and approve this PR. Also don't ask yl33 to fix Input Delegator.

File: docs/application/native/guides/app-management/common-appcontrol.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Reni-90) [reply docs/application/native/guides/app-management/common-appcontrol.md]: OK. Other than this sentence, every changes looks fine.

**Comment by mijin80lee** [comment]
TD Sync Completed.(for Tizen_5.0_prepare)
